lib/k8s/resource_client.rb

```diff
@@ -200,7 +210,30 @@ def list(labelSelector: nil, fieldSelector: nil, namespace: @namespace)
           'fieldSelector' => selector_query(fieldSelector)
         )
       )
-      process_list(list)
+    end
+
+    # @param labelSelector [nil, String, Hash{String => String}]
+    # @param fieldSelector [nil, String, Hash{String => String}]
+    # @param resourceVersion [nil, String]
+    # @param timeout [nil, Integer]
+    # @raise [Excon::Error]
+    def watch(labelSelector: nil, fieldSelector: nil, resourceVersion: nil, timeout: nil, namespace: @namespace)
```

Comment:
Thinking, should this `rescue Excon::Error::Timeout` and maybe collect the watchevents in an array for returning?

In which case, maybe the yield should check for `block_given?` so you don't get `LocalJumpError: no block given` when used without a block.

Reply:
or, how about:

```
wait_for: [:added, :modified, :deleted, :error, :any]
```

Reply:
I don't like collecting the events.. it may cause excessive memory usage. Not sure if I understand the `wait_for`, how should it work?

Reply:
it would exit the `response_block?  when the expected type of event occurs. but maybe the user can implement their own logic and this should be as simple as it now is.